compiler/rustc_resolve/src/errors.rs

```diff
@@ -22,7 +22,7 @@ pub(crate) struct UnderscoreLifetimeNameCannotBeUsedHere(#[primary_span] pub(cra
 
 #[derive(Diagnostic)]
 #[diag(resolve_crate_may_not_be_imported)]
-pub(crate) struct CrateMayNotBeImprted(#[primary_span] pub(crate) Span);
+pub(crate) struct CrateMayNotBeImported(#[primary_span] pub(crate) Span);
```

Comment:
This struct is dead code, no one mentions it. You shouldn't remove it here, but this is interesting anyways. Looks like something from the diagnostics derive is messing up the dead code lint (cc @davidtwco).